### PR TITLE
feat: moving quarkus to reactive

### PR DESCRIPTION
**Pull request by YohannParis**
# Description

* Update `hmi-server` to use reactive development
* Tested, and the app works fine as is.

### CODE REVIEW

**Comment by mwdchang** [comment]
Is it as simple as that?

It seems like aside from dependencies we also need to swap the annotations?
https://quarkus.io/guides/resteasy-reactive-migration

**Comment by YohannParis** [comment]
> Is it as simple as that?
> 
> 
> 
> It seems like aside from dependencies we also need to swap the annotations?
> 
> https://quarkus.io/guides/resteasy-reactive-migration

Yes, there is more work to do. The plan is that this branch will be were all reactive development PR will merge into.

Once we are good we will be able to merge it back to main.

File: packages/services/hmi-server/build.gradle [hunk 1]
```diff
@@ -68,7 +67,7 @@ dependencies {
 }
 
 group "software.uncharted"
-version "1.0.0-SNAPSHOT"
+version "2.0.0"
```
Comment (dvince2) [inline packages/services/hmi-server/build.gradle:70]: Should this stay -SNAPSHOT since this is the development branch?

Comment (YohannParis) [reply packages/services/hmi-server/build.gradle:78]: I don't really now, `main` branch is the actual version we are deploying.

**Review by dvince2 — APPROVED** [review]

**Comment by YohannParis** [comment]
@dvince2 I was able to move to Quarkus 3.2.2, but it can't find `import javax.inject.Inject;` so I'm not able to build the project.

**Comment by YohannParis** [comment]
This is now irrelevant as we move to Spring

**Pull request closed without merging**